extensions/amp-base-carousel/1.0/base-carousel.type.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @externs */
+
+/**
+ * @typedef {{
+ *   arrowNext: (?PreactDef.VNode|undefined),
+ *   arrowPrev: (?PreactDef.VNode|undefined),
+ *   children: (!PreactDef.Renderable),
+ *   defaultSlide: (number|undefined),
+ *   loop: (boolean|undefined),
+ *   slide: (number|undefined),
+ *   onLayout: (function():undefined|undefined),
+ *   onSlideChange: (function(number):undefined|undefined),
+ *   setAdvance: (function(function(number):undefined):undefined|undefined),
+ * }}
+ */
+var BaseCarouselProps;
+
+/**
+ * @typedef {{
+ *   children: !Array<PreactDef.Renderable>,
+ *   ignoreProgrammaticScroll: {current: (boolean|null)},
+ *   loop: (boolean|undefined),
+ *   restingIndex: number,
+ *   setRestingIndex: (function(number):undefined),
+ * }}
+ */
+var ScrollerProps;
```

Comment:
The extern namespace is global. So, unfortunately, we need to be very expansive in our names. E.g. `BaseCarouselScrollerProps` or such. But pinging @jridgewell to confirm. I think closure conflated two things in externs "global namespace" and "type external to compiler".

Reply:
Renamed.